### PR TITLE
Tests for fixed issues in Series[datetime64]

### LINKED ISSUE #12534: Negative timedeltas for deltas > 292 years
#### Code Sample, a copy-pastable example if possible

```
import pandas as pd
dates = pd.Series(pd.date_range('1700-01-01', periods=4500, freq='m'))
days_delta = (dates-dates.min()).astype('timedelta64[D]')
```
#### Expected Output

Should return only positive timedeltas, however for deltas greater than 292 years the values are negative. 
#### output of `pd.show_versions()`

commit: None
python: 3.4.4.final.0
python-bits: 64
OS: Windows
OS-release: 8.1
machine: AMD64
processor: Intel64 Family 6 Model 58 Stepping 9, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None

pandas: 0.17.1
nose: None
pip: 8.0.3
setuptools: 20.1.1
Cython: 0.23.4
numpy: 1.10.4
scipy: 0.17.0
statsmodels: 0.6.1
IPython: 4.0.3
sphinx: 1.3.5
patsy: 0.4.1
dateutil: 2.5.0
pytz: 2015.7
blosc: None
bottleneck: 1.0.0
tables: 3.2.2
numexpr: 2.4.6
matplotlib: 1.5.1
openpyxl: 2.3.2
xlrd: 0.9.4
xlwt: 1.0.0
xlsxwriter: 0.8.4
lxml: 3.5.0
bs4: 4.4.1
html5lib: 0.999
httplib2: None
apiclient: None
sqlalchemy: 1.0.11
pymysql: 0.6.2.None
psycopg2: None
Jinja2: 2.8

### LINKED ISSUE #19071: Series[datetime64tz] - DatetimeIndex[tz] raises ValueError
Opening an issue mostly so the upcoming PR can reference it.  This is in fact already fixed in master (at least as of 93033151a8), so the PR is just a test.

Setup:
```
dti = pd.date_range('1999-09-30', periods=10, tz='US/Pacific')
ser = pd.Series(dti)
```

0.22.0:
```
>>> ser - dti
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/pandas/core/ops.py", line 720, in wrapper
    converted = _Op.get_op(left, right, name, na_op)
  File "/usr/local/lib/python2.7/site-packages/pandas/core/ops.py", line 349, in get_op
    return _TimeOp(left, right, name, na_op)
  File "/usr/local/lib/python2.7/site-packages/pandas/core/ops.py", line 385, in __init__
    self._validate(lvalues, rvalues, name)
  File "/usr/local/lib/python2.7/site-packages/pandas/core/ops.py", line 444, in _validate
    raise ValueError("Incompatible tz's on datetime subtraction "
ValueError: Incompatible tz's on datetime subtraction ops

>>> dti - ser
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/site-packages/pandas/core/ops.py", line 720, in wrapper
    converted = _Op.get_op(left, right, name, na_op)
  File "/usr/local/lib/python2.7/site-packages/pandas/core/ops.py", line 349, in get_op
    return _TimeOp(left, right, name, na_op)
  File "/usr/local/lib/python2.7/site-packages/pandas/core/ops.py", line 385, in __init__
    self._validate(lvalues, rvalues, name)
  File "/usr/local/lib/python2.7/site-packages/pandas/core/ops.py", line 444, in _validate
    raise ValueError("Incompatible tz's on datetime subtraction "
ValueError: Incompatible tz's on datetime subtraction ops
```

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #12534, closes #19071
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by gfyoung** [comment]
@jbrockmendel : To properly close #12534, you should also add documentation per the discussion.

File: pandas/tests/series/test_operators.py [hunk 1]
```diff
@@ -1317,6 +1317,49 @@ def test_datetime64_ops_nat(self):
...
+
+        ser.iloc[1:] = pd.NaT
+        expected = pd.Series(['91279 Days', 'NaT', 'NaT', 'NaT'],
+                             dtype='timedelta64[ns]')
+        res = ser - dt
+        tm.assert_series_equal(res, expected)
+        res = dt - ser
+        tm.assert_series_equal(res, -expected)
+
+    def test_dt64tz_series_sub_dtitz(self):
```
Comment (jreback) [inline pandas/tests/series/test_operators.py]: this seems like a random place to put this test. can you group with other tz type op tests?

**Review by jreback — CHANGES_REQUESTED** [review]
were these fixed by a previous PR? can you reference it

File: pandas/tests/series/test_operators.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/series/test_operators.py]: Sure.

**Comment by jbrockmendel** [comment]
> To properly close #12534, you should also add documentation per the discussion

@gfyoung Can you expand on what you have in mind here?

**Comment by jbrockmendel** [comment]
> were these fixed by a previous PR? can you reference it

For one of the tests I know it was fixed by #19024.  For the other I'm not sure when it got fixed (but certainly no later than 19024)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks, if these issues are not in the whatsnew pls add (new PR).

**Pull request merged**